models/comments.models.js

```diff
@@ -0,0 +1,23 @@
+const db = require('../db/connection')
+const articles = require('../db/data/test-data/articles')
+exports.selectCommentsByArticleId = (article_id) => {
+    let queryStr ='SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id FROM comments WHERE comments.article_id = $1'
+    let count = 0
+    const ValidIdArr = articles.map(()=>{
+        count++
+        return count
+    })
+
+    const idIsValid = ValidIdArr.indexOf(Number(article_id))!==-1
+
+    queryStr += ` ORDER BY created_at;`
+    return db
+      .query(queryStr, [article_id])
+      .then(({rows}) => {
+        if(idIsValid){return rows}
+        if(rows.length===0){return Promise.reject({ message: "article does not exist", status: 404})
+        }
+    console.log(rows)
+        return rows;
+      });
```

Comment:
This model is going to need refactoring. 

Currently, you are using a map to make an array of numbers from 1 to the amount of articles and then checking to see if the id you have passed through is in that array. What about articles that have been deleted? The ids would then be out of numerical order and a valid id that _does_ exist would reject. 

Furthermore you are making this array from the test data which will not be used when you host. This does **not** mean that you should import the dev data.

Your return on line 21 will never be used because _either_ the check is valid and the rows are returned, _or_ it rejects and the status is returned.

I would refer you back to the advanced error handling notes in order to know whether an article exists or not in order for you to select comments for that article. This will absolutely break and error in production and must be fixed before you host.